### DISCUSSION TITLE
Upgrade to latest version

### DISCUSSION THREAD

**Discussion by sarath-tap** [netbox-community/netbox#12080] [General]
Hello team, can someone guide me here how to upgrade to latest version without losing any existing data

**Comment by candlerb** [comment]
Please read and follow https://docs.netbox.dev/en/stable/installation/upgrading/

If you decide to do an upgrade in-situ, always take a backup of your Postgres database before you start: see
https://docs.netbox.dev/en/stable/administration/replicating-netbox/

(You should be taking a backup daily anyway, of course!)

The alternative approach is to build a fresh Netbox instance with the new version, import the postgres database as per the second link above, and then run the upgrade script on the new server (to bring the database up to date). This doesn't touch your original instance at all so is completely safe.

If you are upgrading from a version before 2.11, then you must upgrade to 2.11.12 *first*, then you can upgrade to the latest version.

You should not lose any data. But do read all the release notes for intermediate x.y.0 versions as they will tell you any important changes; in some cases data models change and you'll need to copy data from the old models to the new ones (this happened with site contacts, for example).  But Netbox will refuse to upgrade if you've left data in the old models; the upgrade won't silently delete data.
